tests/YearMonthTest.php

```diff
@@ -330,16 +330,38 @@ public function testWithYear()
         $this->assertYearMonthIs(2001, 5, YearMonth::of(2000, 5)->withYear(2001));
     }
 
+    public function testWithYearIsTheSameYear()
+    {
+        $fixedYear = (int)date('Y');
+        $fixedMonth = (int)date('m');
```

Comment:
Please hardcode the year and month here. Tests need to be reproducible and cannot depend on the current date/time.